File: packages/php/src/NebulaConfigException.php
<?php

declare(strict_types=1);

namespace NebulaToken;

use InvalidArgumentException;

/**
 * A caller mistake: invalid engine configuration (§5), or a device identifier
 * supplied to issue() that is not valid UTF-8 ([N-12]).
 *
 * This is the native error channel of [N-20], never a protocol outcome: no
 * §7 error code is ever raised. It extends InvalidArgumentException so that
 * existing `catch (InvalidArgumentException)` handlers keep working.
 */
final class NebulaConfigException extends InvalidArgumentException
{
    public function __construct(string $message)
    {
        parent::__construct('[NEBULA] ' . $message);
    }
}

File: packages/java/src/main/java/dev/nebulatoken/NebulaConfigException.java
package dev.nebulatoken;

/**
 * A caller mistake: an invalid engine configuration (&sect;5), or an argument
 * that has no valid encoding ([N-12]).
 *
 * <p>This is deliberately not an {@link ErrorCode}. Protocol outcomes describe
 * what a <em>presented token</em> turned out to be; this describes a defect in
 * the calling application, and belongs at the call site where it can be fixed
 * ([N-20]). It extends {@link IllegalArgumentException} so that generic
 * bad-argument handling already in place keeps working.
 *
 * <p>Messages never contain a pepper, a verifier or a raw device identifier
 * ([N-14], [N-46]).
 */
public class NebulaConfigException extends IllegalArgumentException {

    private static final long serialVersionUID = 1L;

    public NebulaConfigException(String message) {
        super("[NEBULA] " + message);
    }
}
